### DISCUSSION TITLE
How to define parallel processes.

### DISCUSSION THREAD

**Discussion by prmdeveloper** [aquasecurity/trivy#4979] [Q&A]
### Question

Hello, currently I would like to perform the FS analysis process as quickly as possible according to the available hardware.
How can I increase the number of files processed in parallel in Trivy? Even if this costs me the consumption of more hardware!

**Example:** 
It's  possible to indicate to Trivy that we would like to run in slow mode!

**--slow                              scan over time with lower CPU and memory utilization**

Why wouldn't it be possible to indicate the number of files to be processed at the same time? or indicate an execution level like (slow, fast, turbo)...
Running more file processes at the same time...

### Target

Container Image

### Scanner

Vulnerability

### Output Format

JSON

### Mode

Standalone

### Operating System

Alpine

### Version

```bash
0.41.1
```

**Comment by simar7** [comment]
Hi I'm curious to hear how large is your input size of files that you are trying to scan? And how long it takes? Some numbers would help.

**Reply by prmdeveloper** [reply]
Imagine that I would like to process approximately 2000 repositories of different sizes, together with the (--scanners vuln,secret,config) modules.
In my tests in this scenario we have a high execution time, with a low consumption of resources.

**Comment by itaysk** [comment]
hi there. trivy is designed to scan a single code repository at a time (or container image for that matter). you can build an automation around trivy to fit it to your need. Aqua's commercial offering is designed to scan entire code or image registries at scale. https://github.com/aquasecurity/resources/blob/main/trivy-aqua.md
